@Test
public
void
testCtThisAccess
(
)
throws
Exception
{
CtType
<
?
>
type
=
build
(
"spoon.test.targeted.testclasses"
,
"InnerClassThisAccess"
)
;
assertEquals
(
"InnerClassThisAccess"
,
type
.
getSimpleName
(
)
)
;
CtMethod
<
?
>
meth1
=
type
.
getElements
(
new
NameFilter
<
CtMethod
<
?
>
>
(
"method2"
)
)
.
get
(
0
)
;
assertEquals
(
"spoon.test.targeted.testclasses.InnerClassThisAccess.this.method()"
,
meth1
.
getBody
(
)
.
getStatements
(
)
.
get
(
0
)
.
toString
(
)
)
;
CtClass
<
?
>
c
=
type
.
getElements
(
new
NameFilter
<
CtClass
<
?
>
>
(
"InnerClass"
)
)
.
get
(
0
)
;
assertEquals
(
"InnerClass"
,
c
.
getSimpleName
(
)
)
;
CtConstructor
<
?
>
ctr
=
c
.
getConstructor
(
type
.
getFactory
(
)
.
Type
(
)
.
createReference
(
boolean
.
class
)
)
;
assertEquals
(
"this.b = b"
,
ctr
.
getBody
(
)
.
getLastStatement
(
)
.
toString
(
)
)
;
}
